src/linked_list/linked_list.js

```diff
@@ -0,0 +1,161 @@
+import { Node } from './index'
+
+export default class LinkedList {
+  constructor() {
+    let _head = null
+    let _size = 0
+
+    this._getHead = function() {
+      return _head
+    }
+    this._setHead = function( node ) {
+      _head = node
+    }
+    this._getSize = function() {
+      return _size
+    }
+    this._incrementSize = function() {
+      _size++
+    }
+    this._decrementSize = function() {
+      _size--
+    }
+    this._resetSize = function() {
+      _size = 0
+    }
+
+    this._findTail = function( node ) {
+      let nextNode = node.next()
+      if ( nextNode === null ) {
+        return node
+      } else {
+        return this._findTail( nextNode )
+      }
+    }
+    this._findNode = function( node, data ) {
+      if ( node === null ) return -1
+      if ( node.data() === data ) {
+        return node
+      } else if ( node.next() === null ) {
+        return -1
+      } else {
+        return this._findNode( node.next(), data )
+      }
+    }
+    this._findPrevNode = function( node, data ) {
+      let nextNode = node.next()
+      if ( nextNode === null ) {
+        return false
+      }
+      if ( nextNode.data() === data ) {
+        return node
+      } else {
+        return this._findPrevNode( nextNode, data )
+      }
+    }
+  }
+
+  insert( data ) {
+    let newNode = new Node( data )
+    let headNode = this.getHeadNode()
+    if ( headNode === null ) {
```

Comment:
yes... in fact I should have used the isEmpty method, woulda made waaaaay more sense.